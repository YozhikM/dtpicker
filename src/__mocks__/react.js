const react = require('react');
// Resolution for requestAnimationFrame not supported in jest error :
// https://github.com/facebook/react/issues/9102#issuecomment-283873039
global.window = global;
window.addEventListener = () => {};
window.requestAnimationFrame = () => {
  throw new Error('requestAnimationFrame is not supported in Node');
};

module.exports = react;

global.spyMount = (component, ...methodNames) => {
  const mount = require('enzyme').mount;
  const wrapper = mount(component);
  const result = [wrapper];
  methodNames.forEach(methodName => {
    result.push(jest.spyOn(wrapper.instance(), methodName));
  });
  wrapper.instance().forceUpdate();
  return result;
};
