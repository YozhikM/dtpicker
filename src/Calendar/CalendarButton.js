/* @flow */

import * as React from 'react';
import SvgIcon from '../SvgIcon/SvgIcon';

type Props = {
  onClick?: Function,
  style?: Object,
};

export default class CalendarButton extends React.Component<Props, void> {
  render() {
    const { onClick, style } = this.props;
    return (
      <div onClick={onClick} style={style}>
        <SvgIcon file="calendar" />
      </div>
    );
  }
}
