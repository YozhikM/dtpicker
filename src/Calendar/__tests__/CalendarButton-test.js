/* @flow */

import * as React from 'react';
import CalendarButton from '../CalendarButton';

describe('CalendarButton', () => {
  describe('render()', () => {
    it('basic', () => {
      const wrapper = shallow(<CalendarButton />);

      expect(wrapper).toMatchSnapshot();
    });

    it('if style', () => {
      const wrapper = shallow(<CalendarButton style={{ backgroundColor: 'rebeccapurple' }} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  it('onClick()', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<CalendarButton onClick={onClick} />);
    wrapper.find('div').simulate('click');

    expect(onClick).toBeCalled();
  });
});
