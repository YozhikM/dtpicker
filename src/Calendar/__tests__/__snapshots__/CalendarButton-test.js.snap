// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarButton render() basic 1`] = `
<div>
  <SvgIcon
    file="calendar"
  />
</div>
`;

exports[`CalendarButton render() if style 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rebeccapurple",
    }
  }
>
  <SvgIcon
    file="calendar"
  />
</div>
`;
